.cicd/scripts/srw_test.sh

```diff
@@ -38,58 +38,15 @@ else
 fi
 
 cd ${we2e_test_dir}
-./setup_WE2E_tests.sh ${platform} ${SRW_PROJECT} ${SRW_COMPILER} ${test_type} \
-    expt_basedir=${we2e_experiment_base_dir} \
-    opsroot=${nco_dir}
-
-# Run the new run_srw_tests script if the machine is Cheyenne.
-if [[ "${platform}" = "cheyenne" ]]; then
-    cd ${workspace}/ush
-    ./run_srw_tests.py -e=${we2e_experiment_base_dir}
-    cd ${we2e_test_dir}
-fi
-
 # Progress file
 progress_file="${workspace}/we2e_test_results-${platform}-${SRW_COMPILER}.txt"
-
-# Allow the tests to start before checking for status.
-# TODO: Create a parameter that sets the initial start delay.
-if [[ "${platform}" != "cheyenne" ]]; then
-    sleep 300
-fi
-
-# Wait for all tests to complete.
-while true; do
-
-    # Check status of all experiments
-    ./get_expts_status.sh expts_basedir="${we2e_experiment_base_dir}" \
-         verbose="FALSE" | tee ${progress_file}
-
-    # Exit loop only if there are not tests in progress
-    set +e
-    grep -q "Workflow status:  IN PROGRESS" ${progress_file}
-    exit_code=$?
-    set -e
-
-    if [[ $exit_code -ne 0 ]]; then
-       break
-    fi
-
-    # TODO: Create a paremeter that sets the poll frequency.
-    sleep 60
-done
-
-# Allow we2e cron jobs time to complete and clean up themselves
-# TODO: Create parameter that sets the interval for the we2e cron jobs; this
-# value should be some factor of that interval to ensure the cron jobs execute
-# before the workspace is cleaned up.
-if [[ "${platform}" != "cheyenne" ]]; then
-    sleep 600
-fi
+./setup_WE2E_tests.sh ${platform} ${SRW_PROJECT} ${SRW_COMPILER} ${test_type} \
+    --expt_basedir=${we2e_experiment_base_dir} \
+    --opsroot=${nco_dir} | tee ${progress_file}
 
```

Comment:
I don't know how the new `run_WE2E` works but just wanted to make sure the various delays are taken care of
- "initial delay" of 300 sec 
- polling frequency of 60 sec
- after completion delay of 600 sec. This one is to make sure jenkins does not delete experiment directory prematurely.

Reply:
Unless I'm missing something, none of those delays should be relevant with the new script. 

1. The `./run_WE2E_tests.py` script takes care of both the setting up and monitoring of tests, so no initial delay is needed
2. The new script serially cycles through active experiments with no delay
3. The new script will not finish until all tasks have been confirmed complete. 

One thing that may need to change is that for larger tests (such as, if the comprehensive tests are re-enabled), it may be worth using the new `-p` parallel option. Because the Jenkins tests call the script with `-d`, calls to rocotorun are fairly slow, taking ~30 seconds each (and there are two calls to rocotorun per test check). But this should be done cautiously depending on the platform: too many parallel tasks may overload the login node if that's where this script is running.

Reply:
For the third one, note that the existing `./run_WE2E_Tests.sh` also does not return until all tasks are complete, so if it was needed for the shell script it may be need for the python script. I vaguely recall that server side jenkins code by @jessemcfarland  deletes experiment directories as soon as `srw_tests.sh` finishes. This could happen before all cronjobs are removed, since they need to be called one last time for them to be removed from the crontable.

The first one, initial delay, is required to given enough time for atleast one workflow to be launched. If `run_WE2E_tests.py` does not exist thinking all jobs completed, if no jobs were launched after say 4 mins, then it should be fine.

The second one is ok i think since it looks like the behaviour of checking experiment status changed. The old `get_expt_status.sh` was time consuming so it checks status of experiments every minute and print to the screen.

Reply:
>note that the existing ./run_WE2E_Tests.sh also does not return until all tasks are complete

The `launch_FV3LAM_wflow.sh` script *tried* to implement this but it was done in quite a hacky way, so there were instances where the script would assume the experiment was done when there were still jobs running (for example, if there was a failed task but other tasks were still running). The new implementation does not use that script (does not use cron jobs either) and so does not suffer from that problem. It will only exit if it has verified that all jobs are done, via reading the rocoto database file directly.

>If run_WE2E_tests.py does not exit thinking all jobs completed, when in reality no jobs were launched after say 4 mins, then it should be fine.

This is correct, the `run_WE2E_tests.py` both submits and monitors jobs, so unless something goes horribly wrong there is no possibility of "desync" there.

I won't promise that I have anticipated *all* potential problems, but if any of these problems you mention (or similar ones) happen in the future, I can fix it in the `run_WE2E_tests.py` script itself, we shouldn't rely on hand-wavy delays to avoid problems anyway.

Reply:
Ok I did not realize cron is no longer being used. I agree the initial/final delays are not required in this case.